modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/AddFormInstanceRecordMVCActionCommand.java

```diff
@@ -104,7 +106,8 @@ protected void doProcessAction(
 		_addFormInstanceMVCCommandHelper.
 			updateRequiredFieldsAccordingToVisibility(
 				actionRequest, ddmForm, ddmFormValues,
-				themeDisplay.getLocale());
+				LocaleUtil.fromLanguageId(
+					LanguageUtil.getLanguageId(actionRequest)));
```

Comment:
@rodrigopaulino, have you tested your whole solution in a published form through its public URL? We had issues related to language specifically on this mentioned scenario.

Reply:
My tests were only using published forms, @natocesarrego. I needed to test the evaluation during the submission process as well.